pkg/migrate/build-manifests.go

```diff
@@ -0,0 +1,267 @@
+package migrate
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/yaml"
+	"strings"
+
+	publicv1 "github.com/cockroachdb/cockroach-operator/apis/v1alpha1"
+	"github.com/cockroachdb/errors"
+	"github.com/cockroachdb/helm-charts/pkg/upstream/cockroach-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type Manifest struct {
+	cloudProvider  string
+	cloudRegion    string
+	object         string
+	namespace      string
+	kubeconfig     string
+	outputDir      string
+	objectManifest string
+}
+
+// Option represents a functional option for configuring Manifest
+type Option func(*Manifest) error
+
+// NewMigration constructs a Manifest with required fields and functional options
+func NewMigration(cloudProvider, cloudRegion, kubeconfig, namespace, outputDir string, opts ...Option) (*Manifest, error) {
```

Comment:
Out of curiosity, what's the value of using a shared `Manifest` type between helm/operator sources? It looks like the two paths share almost zero code, and trying to abstract them leads us to use generic names like `object`, which is a bit confusing.

Reply:
It is because the migration for operator and statefulset are very close.  Later we can only have one path for both migration. This is why I kept the same. In follow up tasks, I am going to simplify this.